qml/Components/Dialogs.qml

```diff
@@ -296,9 +302,11 @@ MouseArea {
         }
 
         onLogoutReady: {
-            root.closeAllApps();
-            Qt.quit();
-            unitySessionService.endSession();
+            topLevelSurfaceList.closeAllWindows();
+            doOnClosedAllWindows = function() {
+                Qt.quit();
```

Comment:
Do we need to actually call `Qt.quit();` here? Shouldn't this happen as a result of the `unitySessionService.endSession();` call instead?

Reply:
It's needed as it's a cleaner shutdown then a call from upstart. Also this makes it work without an unity session.